src/App.js

```diff
@@ -14,11 +17,98 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const preparedGoods = goodsFromServer.map((good, id) => ({
+  name: good,
+  id,
+}));
+
+class App extends PureComponent {
+  state = {
+    isListVisible: false,
+    isButtonVisible: true,
+    goods: preparedGoods,
+    minWordLength: 1,
+  }
+
+  showList = () => {
+    this.setState(state => ({
+      isListVisible: true,
+      isButtonVisible: false,
+    }));
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((prevGood, nextGood) => (
+        prevGood.name.localeCompare(nextGood.name)
+      )),
+    }));
+  };
+
+  resetList = () => {
+    this.setState(state => ({
+      goods: preparedGoods,
+      minWordLength: 1,
+    }));
```

Comment:
```suggestion
    this.setState({
      goods: preparedGoods,
      minWordLength: 1,
    });
```
тебе не нужен prevState